app.js

```diff
@@ -14,3 +14,98 @@ const books = [
   { author: "Aldous Huxley", title: "Animal Farm", id: 9 },
   { author: "Aldous Huxley", title: "Brave New World", id: 10 },
 ];
+
+class Patron {
+  constructor(name, id) {
+    this.name = name;
+    this.id = id;
+  }
+}
+class BookStatus {
+  constructor(checkedOut = false, Patron = null) {
+    this.checkedOut = checkedOut;
+    this.Patron = Patron;
+  }
+}
+
+class Book {
+  constructor(title, author, id) {
+    this.title = title;
+    this.author = author;
+    this.id = id;
+    this.status = new BookStatus();
+  }
+}
+class Library {
+  constructor(name, books) {
+    this.name = name;
+    this.books = books;
+  }
+  //يضيف عنصر الى المصفوفة
+  addBook(title, author, id) {
+    this.books.push(new Book(title, author, id));
+  }
+  // يحذف عنصر من المصفوفة
+  removeBook(id) {
+    this.books = this.books.filter(() => Book.id !== id);
+  }
+
+  // يجيب ال id من المصفوفة
+  getBookById(id) {
+    const found = this.books.find((book) => book.id === id);
+    return found;
+  }
+  //تفحص الكتاب المأخوذ من المصفوفة
+  checkOutBook(id, patron) {
+    const book = this.getBookById(id);
+
+    if (!book) {
+      return;
+    }
+    if (!book.status) {
+      book.status = new BookStatus(patron, true);
+      return;
+    }
+    if (book.status.checkedOut) {
+      return;
+    }
+    book.status.checkedOut = true;
+    book.status.patron = patron;
+  }
+  //ترجع او نفحص الكتاب المأخوذ
+  checkInBook(id) {
+    const book = this.getBookById(id);
+    book.status.checkOutBook = false;
```

Comment:
There's no property in the book status called `checkOutBook`, there's only `checkedOut` thus this should be written as:
```
this.status.checkedOut = false
```